src/llama_stack/providers/remote/prompts/mlflow/mlflow.py

```diff
@@ -0,0 +1,547 @@
+    """
+
+    def __init__(self, config: MLflowPromptsConfig):
+        """Initialize MLflow prompts adapter.
+
+        Args:
+            config: MLflow provider configuration
+        """
+        self.config = config
+        self.mlflow_client: "MlflowClient | None" = None
+        self.mapper = PromptIDMapper()
+        logger.info(
+            f"MLflowPromptsAdapter initialized: tracking_uri={config.mlflow_tracking_uri}, "
+            f"experiment={config.experiment_name}"
+        )
+
+    def _get_mlflow_token(self) -> str | None:
+        """Get MLflow API token from provider data or config.
+
+        Provider data takes precedence over configuration, following the pattern
+        used by inference providers for credential management.
+
+        Returns:
+            MLflow API token if available, None otherwise
+        """
+        # First try to get from config
+        token = None
+        if self.config.auth_credential is not None:
+            token = self.config.auth_credential.get_secret_value()
+
+        # Provider data overrides config (per-request credentials)
+        provider_data = self.get_request_provider_data()
+        if provider_data and provider_data.mlflow_api_token:
+            token = provider_data.mlflow_api_token
+            logger.debug("Using MLflow token from provider data header")
+
+        return token
+
+    async def initialize(self) -> None:
+        """Initialize MLflow client and set up experiment.
+
+        Sets up MLflow connection with optional authentication via token.
+        Token can be provided via config or will be read from environment variables
+        (MLFLOW_TRACKING_TOKEN, etc.) as per MLflow's standard behavior.
+
+        Raises:
+            ImportError: If mlflow package is not installed
+            Exception: If MLflow connection fails
+        """
+        if mlflow is None:
+            raise ImportError(
+                "mlflow package is required for MLflow prompts provider. "
+                "Install with: pip install 'mlflow>=3.4.0'"
+            )
+
+        # Set MLflow URIs
+        mlflow.set_tracking_uri(self.config.mlflow_tracking_uri)
+
+        if self.config.mlflow_registry_uri:
+            mlflow.set_registry_uri(self.config.mlflow_registry_uri)
+        else:
+            # Default to tracking URI if registry not specified
+            mlflow.set_registry_uri(self.config.mlflow_tracking_uri)
+
+        # Set authentication token if provided in config
+        # Note: Provider data tokens are handled per-request, not at initialization
+        if self.config.auth_credential is not None:
+            import os
+
+            # MLflow reads MLFLOW_TRACKING_TOKEN from environment
+            os.environ["MLFLOW_TRACKING_TOKEN"] = self.config.auth_credential.get_secret_value()
+            logger.debug("Set MLFLOW_TRACKING_TOKEN from config auth_credential")
+
+        # Initialize client
+        self.mlflow_client = MlflowClient()
```

Comment:
this is unused.

Reply:
This is used in the edge case of managing prompts created outside Llama Stack. I'll update the code for clarity.